### ISSUE TITLE
Trouble Building

### ISSUE THREAD

**Issue by AnonymousAJ** [AdmiralCurtiss/HyoutaTools#19]
I’ve installed the latest versions of .NET, downloaded the three submodules/utilities and put them in their correct folders, and then tried building HyoutaToolsCLI on both VS 2020 and VS 2017 (using fresh installs of HyoutaTools when attempting each, save for adding in the submodules again), but I’ve run into issues with both versions.

When trying to build it on VS 2017, I’m completely stopped by it giving me these errors saying that I don’t have the right version of .NET 5.0, even though I know I have it installed:
![VS 2017 Errors](https://user-images.githubusercontent.com/48224502/166621500-fb7bc61f-c651-470e-9675-9f19268844f3.png)

And when I try to build it on VS 2020, I get slightly further and am able to start building, but it then throws these errors and doesn’t finish building properly (A Bin folder is created in HyoutaToolsCLI, but the folders inside it are empty):
![Build Errors](https://user-images.githubusercontent.com/48224502/166622273-33f5cfb0-aa02-44ca-b9c6-59e4785b8841.png)

Here is the Output text for the attempt using VS 2020, as well:
![Build Output](https://user-images.githubusercontent.com/48224502/166621527-4939cad6-d1f7-4a3c-9639-220c560c28a2.png)

I’m not sure what to do from here—I’m fairly new with GitHub and building Projects in VS, so I might just be missing a step or two, here, so if that’s the case then please let me know what I’m missing/doing wrong

**Comment by AdmiralCurtiss** [comment]
Well first off, the way to get the code is to run

~~~
git clone --recursive https://github.com/AdmiralCurtiss/HyoutaTools.git
~~~

This automatically fetches all the submodules. If you've already cloned you can instead do

~~~
git submodule update --init --recursive
~~~

---

As for the build failure parts, I'm not sure what's happening there. I just tried building a fresh copy with both VS2019 and VS2022 and they just work out of the box for me. I already have the .NET SDKs installed obviously, but I think you need the .NET 5.0 one from [here](https://dotnet.microsoft.com/en-us/download/visual-studio-sdks) for VS2019 and... possibly the same one for VS2022? Not sure about that one. VS2017 is probably too old, and VS2020 doesn't exist so I assume you meant 2022.

**Comment by AdmiralCurtiss** [comment]
For VS2022 it might be this mysterious SDK in the VS installer:

![Clipboard02](https://user-images.githubusercontent.com/4522237/166640766-a251c0af-3e7e-490b-83e8-67da8c85c7b3.png)

**Comment by AnonymousAJ** [comment]
Ahh, okay, so I was just way off on how to download it, then. I originally got each of the components/submodules by downloading each of them as .ZIP files, then piecing them together from there, but once I ran the first line of code you gave me instead, it's now working and building perfectly with no errors!
Thank you very much for the help and quick reply! I'll close up this issue now, then

(Side note, not sure how I didn't notice that I kept writing 2020 instead of 2022...)
